### PR TITLE
[dev] Adding a switch to skip chroot rebuilds

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Due to how packages are built, the worker chroot kept being constantly rebuilt, even if it was unnecessary. Fixing the underlying issue with proper detection of when the rebuild is required will require more complex changes to our build infrastructure, so for now I'm adding the `REFRESH_WORKER_CHROOT` argument, to skip rebuilds, if you know they are not necessary. The default behaviour remains unchanged.

###### Change Log  

- Adding an new `REFRESH_WORKER_CHROOT` build tools argument.
- 
###### Does this affect the toolchain?  

No.

###### Test Methodology

- Local package builds.

### CODE REVIEW

File: toolkit/scripts/tools.mk [hunk 1]
```diff
@@ -135,16 +135,18 @@ worker_chroot_manifest = $(TOOLCHAIN_MANIFESTS_DIR)/$(worker_manifest_name)
...
-sed_regex_arch_only = 's`(.*\.([^\.]+)\.rpm)`\2`p'
 worker_chroot_rpm_paths := $(shell sed -nr $(sed_regex_full_path) < $(worker_chroot_manifest))
 
 worker_chroot_deps := \
 	$(worker_chroot_manifest) \
 	$(worker_chroot_rpm_paths) \
-	$(toolchain_rpms) \
 	$(PKGGEN_DIR)/worker/create_worker_chroot.sh
 
+ifeq ($(REFRESH_WORKER_CHROOT),y)
```
Comment (mfrw) [inline toolkit/scripts/tools.mk:145]: +1 A welcome change :)
Thank you!

File: toolkit/docs/building/building.md [hunk 2]
```diff
@@ -487,6 +490,20 @@ sudo make hydrate-rpms PACKAGE_ARCHIVE=./rpms.tar.gz
...
 
+#### `REFRESH_WORKER_CHROOT=...`
+
+##### `REFRESH_WORKER_CHROOT=`**`n`**
+
+> If exists, don't attempt to rebuild the worker chroot, even if the its build script, its manifest, or the packages it consists of have been modified in the local repository.
+
+##### `REFRESH_WORKER_CHROOT=`**`y`** *(default)*
+
+> Rebuild the worker chroot every time at least one of the following has changes:
```
Comment (mfrw) [inline toolkit/docs/building/building.md:501]:
```suggestion
> Rebuild the worker chroot every time if at least one of the following has changes:
```

File: toolkit/docs/building/building.md [hunk 3]
```diff
@@ -487,6 +490,20 @@ sudo make hydrate-rpms PACKAGE_ARCHIVE=./rpms.tar.gz
 
 > Build the go tools from source as needed.
 
+#### `REFRESH_WORKER_CHROOT=...`
+
+##### `REFRESH_WORKER_CHROOT=`**`n`**
+
+> If exists, don't attempt to rebuild the worker chroot, even if the its build script, its manifest, or the packages it consists of have been modified in the local repository.
```
Comment (mfrw) [inline toolkit/docs/building/building.md:497]:
```suggestion
> If exists, don't attempt to rebuild the worker chroot, even if it is the build script, its manifest, or the packages it consists of have been modified in the local repository.
```

**Review by mfrw — APPROVED** [review]
A few nits, otherwise LGTM :)

File: toolkit/docs/building/building.md (hunk 2 above)
Comment (PawelWMS) [reply toolkit/docs/building/building.md]: I believe this is correct, but I've noticed a typo in "changes" - I wanted to say "changed". Thanks!

File: toolkit/docs/building/building.md (hunk 3 above)
Comment (PawelWMS) [reply toolkit/docs/building/building.md]: Another typo, thank you! Removed "the".:)

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**